pkg/controller/operators/catalog/operator.go

```diff
@@ -1028,6 +1034,80 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		return err
 	}
 
+	// Attempt to unpack bundles before installing
+	// Note: This should probably use the attenuated client to prevent users from resolving resources they otherwise don't have access to.
+	if len(bundleLookups) > 0 {
+		logger.Debug("unpacking bundles")
+
+		var unpacked bool
+		unpacked, steps, bundleLookups, err = o.unpackBundles(namespace, steps, bundleLookups, unpackTimeout)
+		if err != nil {
+			// If the error was fatal capture and fail
+			if olmerrors.IsFatal(err) {
```

Comment:
Sorry for stupid question, but is this type of failure the 'fatal' one?

```yaml
bundle unpacking failed. Reason: DeadlineExceeded, and Message: Job was
        active longer than specified deadline
```

Reply:
No, failure from your example will fall under `if isFailed` branch.

Here is an example of `FatalError`s:

https://github.com/operator-framework/operator-lifecycle-manager/blob/858c7cdb2f4cfe3d351225972aede54e3b114ce4/pkg/controller/registry/resolver/steps.go#L122-L129